cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/AbstractMapReduceContextBuilder.java

```diff
@@ -109,8 +112,8 @@ public BasicMapReduceContext build(MapReduceMetrics.TaskType type,
     MapReduceSpecification spec = program.getApplicationSpecification().getMapReduce().get(program.getName());
     BasicMapReduceContext context =
       new BasicMapReduceContext(program, type, RunIds.fromString(runId), taskId, runtimeArguments, datasets, spec,
-                                logicalStartTime, workflowBatch, discoveryServiceClient, metricsCollectionService,
-                                datasetFramework, adapterSpec, pluginInstantiator);
+                                logicalStartTime, workflowBatch, workflowToken, discoveryServiceClient,
```

Comment:
`workflowBatch` carries the name of the current program which Workflow wants to run, which is not included in the `WorkflowContext`.